### PR TITLE
[Doc] [Enhancement] Add more clarification to initial value documentation for form

### PR DESCRIPTION
**Pull request by mnayef95**
[talabat.com](https://www.talabat.com) [Talabat Flutter PRs](https://github.com/search?q=org%3Aflutter+talabat&type=pullrequests)

When you change the `initialValue` of a `TextFormField`, it will not trigger a widget rebuild, which is the intended behavior. This PR is meant to clarify that. https://github.com/flutter/flutter/issues/152816

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter/lib/src/widgets/form.dart [hunk 1]
```diff
@@ -510,6 +510,7 @@ class FormField<T> extends StatefulWidget {
   ///
   /// This is called `value` in the [DropdownButtonFormField] constructor to be
   /// consistent with [DropdownButton].
+  /// The `initialValue` only affects the initial state when the form field is first built.
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/form.dart:513]: Can you put an empty line of `///` above this?

Comment (justinmc) [reply packages/flutter/lib/src/widgets/form.dart:513]: Oh also, it affects the state when FormField.reset is called!

**Review by justinmc — COMMENTED** [review]
I appreciate you helping to improve the docs! Just some small nits.

File: packages/flutter/lib/src/widgets/form.dart (hunk 1 above)
Comment (mnayef95) [reply packages/flutter/lib/src/widgets/form.dart]: @justinmc Thank you for the feedback, addressed it

**Review by justinmc — APPROVED** [review]
LGTM 👍 Thanks for making those changes.

File: packages/flutter/lib/src/widgets/form.dart [hunk 2]
```diff
@@ -510,6 +510,10 @@ class FormField<T> extends StatefulWidget {
   ///
   /// This is called `value` in the [DropdownButtonFormField] constructor to be
   /// consistent with [DropdownButton].
+  ///
+  /// The `initialValue` affects the form field's state in two cases:
+  /// 1. When the form field is first built, `initialValue` determines the field's initial state.
+  /// 2. When [FormField.reset] is called, the form field is reset to this `initialValue`.
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/form.dart:516]: Ah this should be [FormState.reset]. That's why the docs test is failing.

Comment (nate-thegrate) [reply packages/flutter/lib/src/widgets/form.dart:516]: Yes—or I guess more specifically, `[FormFieldState.reset]` is the direct way to do it; `FormState.reset()` calls this method for all of the descendant fields.

```suggestion
  /// 2. When [FormFieldState.reset] is called (either directly or by calling
  ///    [FormState.reset]), the form field is reset to this `initialValue`.
```

**Review by nate-thegrate — APPROVED** [review]
LGTM, pending the update based on Justin's comment :)

File: packages/flutter/lib/src/widgets/form.dart (hunk 2 above)
Comment (justinmc) [reply packages/flutter/lib/src/widgets/form.dart]: You're right, good catch!

**Pull request merged**